docs/src/man/basics.md

```diff
@@ -0,0 +1,1505 @@
+  997 │   996     40  male        3  own      little           little                     3857        30  car
+  998 │   997     38  male        2  own      little           NA                          804        12  radio/TV
+  999 │   998     23  male        2  free     little           little                     1845        45  radio/TV
+ 1000 │   999     27  male        2  own      moderate         moderate                   4576        45  car
+                                                                                                              985 rows omitted
+```
+
+You can also compute descriptive statistics directly on individual columns:
+
+```jldoctest dataframe
+julia> using Statistics
+
+julia> mean(german.Age)
+35.546
+```
+
+`mapcols` function return a `DataFrame` where each column of `german` is transformed using function `f`.
+Note that `mapcols` guarantees not to reuse the columns from `german` in the returned `DataFrame`. If `f` 
+returns its argument then it gets copied before being stored. 
+
+```jldoctest dataframe
+julia> german = copy(german_ref);
+
+julia> mapcols(id -> id.^2, german)
+1000×10 DataFrame
+  Row │ id      Age    Sex           Job    Housing   Saving accounts       Ch ⋯
+      │ Int64   Int64  String        Int64  String    String                St ⋯
+──────┼─────────────────────────────────────────────────────────────────────────
+    1 │      0   4489  malemale          4  ownown    NANA                  li ⋯
+    2 │      1    484  femalefemale      4  ownown    littlelittle          mo
+    3 │      4   2401  malemale          1  ownown    littlelittle          NA
+    4 │      9   2025  malemale          4  freefree  littlelittle          li
+    5 │     16   2809  malemale          4  freefree  littlelittle          li ⋯
+    6 │     25   1225  malemale          1  freefree  NANA                  NA
+    7 │     36   2809  malemale          4  ownown    quite richquite rich  NA
+    8 │     49   1225  malemale          9  rentrent  littlelittle          mo
+  ⋮   │   ⋮       ⋮         ⋮          ⋮       ⋮               ⋮               ⋱
+  994 │ 986049    900  malemale          9  ownown    littlelittle          li ⋯
+  995 │ 988036   2500  malemale          4  ownown    NANA                  NA
+  996 │ 990025    961  femalefemale      1  ownown    littlelittle          NA
+  997 │ 992016   1600  malemale          9  ownown    littlelittle          li
+  998 │ 994009   1444  malemale          4  ownown    littlelittle          NA ⋯
+  999 │ 996004    529  malemale          4  freefree  littlelittle          li
+ 1000 │ 998001    729  malemale          4  ownown    moderatemoderate      mo
+                                                  4 columns and 985 rows omitted
+```
+
+If you want to look at first and last rows of a data frame (respectively) then you can do this
+using `first` and `last` functions:
+
+```jldoctest dataframe
+julia> german = copy(german_ref);
```

Comment:
why copy?

Reply:
Previously I had performed an operation which had changed our data frame.

Reply:
OK

Reply:
> Previously I had performed an operation which had changed our data frame.

I have checked and could not find this operation. Can you please point me to it?

Reply:
resolved.